### ISSUE TITLE
Add support for the MPRIS2 protocol

### ISSUE THREAD

**Issue by JustABanana** [MusicPlayerDaemon/MPD#302]
It'd be great if we could have MPRIS2 as a output device to control MPD and display info

**Comment by JustABanana** [comment]
Does unstable label mean it got already implemented and should appear on stable relaxed soon?

**Comment by MaxKellermann** [comment]
No. It means it would need to be implemented in the unstable branch, if ever. Nobody is working on that, and I know nobody who's interested in implementing it. This is only the second time in 10 years MPRIS was ever mentioned. Given how complicated D-Bus and MPRIS is (and how unnecessary this complexity is), I don't think it's worth my time.

**Comment by JustABanana** [comment]
All right then. Thanks for amazing app

**Comment by carnager** [comment]
@JustABanana Take a look at https://github.com/eonpatapon/mpDris2

**Comment by evgfilim1** [comment]
One more good thing that worth looking at is https://github.com/natsukagami/mpd-mpris

**Comment by leigh123linux** [comment]
cantata (mpd client) has mpris support builtin.

**Comment by coxackie** [comment]
I would be nice to have this support actually built-in in `mpd`. Is it quite difficult to port the go code from https://github.com/natsukagami/mpd-mpris?

**Comment by MaxKellermann** [comment]
No Go code will be merged into MPD. For a simple feature like this one, it would add a huge amount of overhead (the whole Go runtime with its garbage collector and dozens of threads each consuming megabytes of RAM), and would complicate the MPD build.

**Comment by coxackie** [comment]
@MaxKellermann  Sorry, I think I got misunderstood. I meant to ask how hard it would be to port the specific go code to c. Not to actually use the code as is.
